samza-sql/src/main/java/org/apache/samza/sql/avro/AvroTypeFactoryImpl.java

```diff
@@ -60,62 +59,68 @@ private SqlSchema convertSchema(List<Schema.Field> fields) {
 
     SqlSchemaBuilder schemaBuilder = SqlSchemaBuilder.builder();
     for (Schema.Field field : fields) {
-      SqlFieldSchema fieldSchema = convertField(field.schema());
+      // Consider any field with default value as nullable. Is it the right assumption ?
```

Comment:
This is slightly tricky. Nullable fields in avro is of type union with one of the value being null. 
The fields with default value means they are optional during serializing and deserializing. 

The behavior on the consumption side. 
1. All avro fields should have values including nullable fields as well as optional fields(fields with default values)

Behavior on the producer side.
1. fields with default values are optional
2. Nullable fields may still need values if they are not optional

So nullability and optionality are slightly different. I am not sure it is a good idea to conflate them.

Reply:
Thanks for clarifying the behavior. Updated the logic by separating out nullable fields and fields with default values. Using default values for validation but not nullable fields.